python/lsst/ts/ess/ess_csc.py

```diff
@@ -135,21 +141,29 @@ async def read(self) -> dict:
         data : `dict`
             A dictionary with objects representing the string read.
         """
+        if not self.reader or not self.connected:
+            raise RuntimeError("Not connected")
+
         read_bytes = await asyncio.wait_for(
             self.reader.readuntil(tcpip.TERMINATOR), timeout=SOCKET_TIMEOUT
         )
         data = json.loads(read_bytes.decode())
         return data
 
-    async def write(self, **data: dict) -> None:
-        """Write the data appended with a newline character.
+    async def write(self, command: str, **data: dict) -> None:
```

Comment:
The type of `**kwargs` dicts (which you are spelling as `**data`, but it's the same thing) is the type of one of the values, often `str` or `typing.Any`. Similarly the type of `*args` is the type of one of the items.

If the values in `data` are not typically dicts then please fix this and look for other instances.

Reply:
Either a configuration dict or an empty dict is passed so I changed it to `Optoinal[Any]`. MyPy is happy with that.

Reply:
Optional implies None, which is not one of the options. Does `Any` work?

Reply:
You're right of course! `Any` works so I've changed it to that. Thanks!